contrib/mesos/pkg/scheduler/components/algorithm/podschedulers/fcfs.go

```diff
@@ -82,25 +60,44 @@ func (fps *fcfsPodScheduler) SchedulePod(r offers.Registry, task *podtask.T) (of
 			return false, nil
 		}
 
-		if fps.FitPredicate()(task, offer, n) {
-			if p.Acquire() {
-				acceptedOffer = p
-				log.V(3).Infof("Pod %s accepted offer %v", podName, offer.Id.GetValue())
-				return true, nil // stop, we found an offer
-			}
+		spec := &podtask.Spec{}
+		err := fps.procurement.Procure(task, offer, n, spec)
```

Comment:
offer.Resources may be mutated by Procurement. If any part of the procurement chain fails (returns error) and the offer is left in the pool (to be re-matched against some later pod), it may have been irreparably modified by the procurement attempt.